src/kubernetes_resources.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+import kubernetes
```

Comment:
So I didin't feel entirely comfortable with this approach of creating K8s Resources and tried it out with lightkube.

[Here](https://github.com/canonical/training-operator/tree/lightkube) is a branch which shows how this would look like with lightkube and just plain manifests. I added some templating just to see if it works and it does. Of course this is just something I put together in an hour so feel free to change it and add more templating.

I think we should use this approach instead of the current one. It seems a lot more maintainable in the long run and more similar to what we will get with Juju Templates (I hope).

Pinging @jnsgruk @knkski and @ca-scribner so that they take a look.

Reply:
I'm pretty encouraged as well.  I'm interested in playing with their templating too as that could be very nice